assignments/tic-tac-toe/code.js

```diff
@@ -5,6 +5,164 @@
 
+let player = 'x'
+let board = ['', '', '', '', '', '', '', '', ''];
+
 registerOnclick((x, y) => {
-  drawText('X', x, y, 'black', Math.min(width, height) * 0.3);
+  ticTacToe(x, y)
 });
+
+drawLine(width / 3, 0, width / 3, height, 'black', 5)
+drawLine(width / 3 * 2, 0, width / 3 * 2, height, 'black', 5)
+drawLine(0, height / 3, width, height / 3, 'black', 5)
+drawLine(0, height / 3 * 2, width, height / 3 * 2, 'black', 5)
+
+const topLeft = (x, y) => {
+  return x < width / 3 && y < height / 3
+}
+const topMid = (x, y) => {
+  return x > width / 3 && x < width / 3 * 2 && y < height / 3
+}
+const topRight = (x, y) => {
+  return x > width / 3 * 2 && y < height / 3
+}
+const midLeft = (x, y) => {
+  return x < width / 3 && y > height / 3 && y < height / 3 * 2
+}
+const midMid = (x, y) => {
+  return x < width / 3 * 2 && x > width / 3 && y > height / 3 && y < height / 3 * 2
+}
+const midRight = (x, y) => {
+  return x > width / 3 * 2 && y > height / 3 && y < height / 3 * 2
+}
+const botLeft = (x, y) => {
+  return x < width / 3 && y > height / 3 * 2
+}
+const botMid = (x, y) => {
+  return x < width / 3 * 2 && x > width / 3 && y > height / 3 * 2
+}
+const botRight = (x, y) => {
+  return x > width / 3 * 2 && y > height / 3 * 2
+}
+
+const ticTacToe = (x, y) => {
+  if (topLeft(x, y) && (board[0] !== ('x' && 'o'))) {
+    drawText(player, width / 6, height / 12 * 3, 'black', 70)
+    board[0] = player
+  } else if (topMid(x, y) && (board[1] !== ('x' && 'o'))) {
+    drawText(player, width / 2, height / 12 * 3, 'black', 70)
+    board[1] = player
+  } else if (topRight(x, y) && (board[2] !== ('x' && 'o'))) {
+    drawText(player, width / 6 * 5, height / 12 * 3, 'black', 70)
+    board[2] = player
+  } else if (midLeft(x, y) && (board[3] !== ('x' && 'o'))) {
+    drawText(player, width / 6, height / 12 * 7, 'black', 70)
+    board[3] = player
+  } else if (midMid(x, y) && (board[4] !== ('x' && 'o'))) {
+    drawText(player, width / 2, height / 12 * 7, 'black', 70)
+    board[4] = player
+  } else if (midRight(x, y) && (board[5] !== ('x' && 'o'))) {
+    drawText(player, width / 6 * 5, height / 12 * 7, 'black', 70)
+    board[5] = player
+  } else if (botLeft(x, y) && (board[6] !== ('x' && 'o'))) {
+    drawText(player, width / 6, height / 12 * 11, 'black', 70)
+    board[6] = player
+  } else if (botMid(x, y) && (board[7] !== ('x' && 'o'))) {
+    drawText(player, width / 2, height / 12 * 11, 'black', 70)
+    board[7] = player
+  } else if (botRight(x, y) && (board[8] !== ('x' && 'o'))) {
+    drawText(player, width / 6 * 5, height / 12 * 11, 'black', 70)
+    board[8] = player
+  }
+  if (player === 'x') {
+    player = 'o'
+  } else {
+    player = 'x'
+  }
+  console.log(board)
+}
+
+const win = () => {
+  return winner || boardFull
+  //drawLine(width / 12, height / 6, width / 12 * 11, height / 6)
+}
+
+const boardFull = () => {
+  return board.every((x) => x !== '')
+}
+
+const winner = () => {
+  return rowWin || columnWin || diagonalWin
+}
```

Comment:
Following the pattern I discussed above, I'd change this to take an argument which I'd then pass down to each of the called functions (which would need to be changed to accept the board as an argument):

```suggestion
const winner = (b) => {
  return rowWin(b) || columnWin(b) || diagonalWin(b)
}
```